### ISSUE TITLE
Problem with function hoisting

### ISSUE THREAD

**Issue by lucassus** [schmod/babel-plugin-angularjs-annotate#29]
Input code:

```javascript
module.exports = function($filterProvider) {
  'ngInject';

  function ordinalSuffixFilter(ordinalSuffix) {
    'ngInject';


    return function(day) {
      if (angular.isNumber(day) && isFinite(day)) {
        return ordinalSuffix(Number(day));
      }
    };
  }

  $filterProvider.register('sqOrdinalSuffix', ordinalSuffixFilter);

};
```

Output code: 

```javascript
'use strict';

ordinalSuffixFilter.$inject = ['ordinalSuffix'];
module.exports = ['$filterProvider', function ($filterProvider) {
  'ngInject';

  function ordinalSuffixFilter(ordinalSuffix) {
    'ngInject';

    return function (day) {
      if (angular.isNumber(day) && isFinite(day)) {
        return ordinalSuffix(Number(day));
      }
    };
  }

  $filterProvider.register('sqOrdinalSuffix', ordinalSuffixFilter);
}];
```

In the given example `ordinalSuffixFilter.$inject = ['ordinalSuffix'];` is defined on the wrong scope therefore it leads to "undefined variable orfinalSuffixFilter" error. I believe that in order to avoid such errors annotations should be added directly above the annotated function.

**Comment by lucassus** [comment]
It could be related to https://github.com/schmod/babel-plugin-angularjs-annotate/issues/22

**Comment by schmod** [comment]
I'll take a look at this.  For now, you can work around this by declaring the function as a variable.

**Comment by schmod** [comment]
Fixed in v0.8.2
